lib/ClassMover/Adapter/WorseTolerant/WorseTolerantMemberFinder.php

```diff
@@ -377,7 +377,8 @@ private function attachClassInfoToReference(MemberReference $reference, ClassMem
             return $reference;
         }
 
-        $accepts = $type->accepts(TypeFactory::class((string) $query->class()));
+
+        $accepts = TypeFactory::reflectedClass($this->reflector, (string)$query->class())->accepts($type);
```

Comment:
index search: this was the wrong way round, the _queried_ class should accept the "found" class.